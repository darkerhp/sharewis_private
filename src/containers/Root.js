/* global __DEV__ */
import React, { Component } from 'react';
import { AsyncStorage } from 'react-native';
import { LoginManager } from 'react-native-fbsdk';
import { Provider } from 'react-redux';
import { persistStore } from 'redux-persist';
import I18n from 'react-native-i18n';
import Orientation from 'react-native-orientation';
import SleekLoadingIndicator from 'react-native-sleek-loading-indicator';

import setupI18n from '../utils/translations';
import PURGE_STORAGE from '../constants/Debug';
import configureStore from '../store';
import App from './App';

const store = configureStore();
setupI18n();


export default class Root extends Component {
  constructor() {
    super();
    this.state = { rehydrated: false };
  }

  componentWillMount() {
    const persistor = persistStore(
      store,
      { storage: AsyncStorage },
      () => this.setState({ rehydrated: true }),
    );
    if (__DEV__ && PURGE_STORAGE) {
      LoginManager.logOut();
      persistor.purge();
    }
  }

  componentDidMount() {
    Orientation.lockToPortrait();
  }

  render() {
    if (!this.state.rehydrated) {
      return <SleekLoadingIndicator loading text={I18n.t('loading')} />;
    }

    return (
      <Provider store={store}>
        <App />
      </Provider>
    );
  }
}
