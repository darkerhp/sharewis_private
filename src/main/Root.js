/* global __DEV__ */
import React, { Component } from 'react';
import { AsyncStorage } from 'react-native';
import { Client } from 'bugsnag-react-native';
import { LoginManager } from 'react-native-fbsdk';
import { Provider } from 'react-redux';
import { persistStore } from 'redux-persist';
import I18n from 'react-native-i18n';
import Orientation from 'react-native-orientation';

import setupI18n from '../config/locales';
import setupMixpanel from '../config/mixpanel';
import PURGE_STORAGE from '../config';
import configureStore from '../redux/store';
import transform from '../lib/transform';
import App from './App';

import SleekLoadingIndicator from '../components/SleekLoadingIndicator';

const bugsnag = new Client();
const store = configureStore();
setupI18n();
setupMixpanel();

export default class Root extends Component {
  constructor() {
    super();
    this.state = { rehydrated: false };
  }

  componentWillMount() {
    const persistor = persistStore(
      store,
      {
        storage: AsyncStorage,
        blacklist: ['ui', 'routes', 'form'],
        transforms: [transform]
      },
      () => {
        this.setState({ rehydrated: true });
      }
    );

    if (__DEV__ && PURGE_STORAGE) {
      LoginManager.logOut();
      persistor.purge();
    }
  }

  componentDidMount() {
    Orientation.lockToPortrait();
  }

  render() {
    if (!this.state.rehydrated) {
      return <SleekLoadingIndicator loading text={I18n.t('loading')} />;
    }

    return (
      <Provider store={store}>
        <App />
      </Provider>
    );
  }
}
