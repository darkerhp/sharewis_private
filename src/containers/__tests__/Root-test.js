import 'react-native';
// Note: test renderer must be required after react-native.
/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
import renderer from 'react-test-renderer';
import React from 'react';
import Root from '../Root';

/* global describe, it, expect, jest */
jest.mock('ActivityIndicatorIOS', (): ReactComponent => 'ActivityIndicatorIOS');
jest.mock('ProgressBarAndroid', (): ReactComponent => 'ProgressBarAndroid');
jest.mock('TouchableNativeFeedback', (): ReactComponent => 'TouchableNativeFeedback');
jest.mock('TextInput', (): ReactComponent => 'TextInput');

describe('Root', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Root />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
