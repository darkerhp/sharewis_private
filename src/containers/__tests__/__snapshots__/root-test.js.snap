exports[`Root renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden"
      },
      Object {
        "backgroundColor": "white",
        "flex": 1
      }
    ]
  }>
  <View
    onMoveShouldSetResponder={[Function onMoveShouldSetResponder]}
    onMoveShouldSetResponderCapture={[Function onMoveShouldSetResponderCapture]}
    onResponderEnd={[Function onResponderEnd]}
    onResponderGrant={[Function onResponderGrant]}
    onResponderMove={[Function onResponderMove]}
    onResponderReject={[Function onResponderReject]}
    onResponderRelease={[Function onResponderRelease]}
    onResponderStart={[Function onResponderStart]}
    onResponderTerminate={[Function onResponderTerminate]}
    onResponderTerminationRequest={[Function bound _handleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function onStartShouldSetResponder]}
    onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
    onTouchStart={[Function bound _handleTouchStart]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden"
      }
    }>
    <View
      onStartShouldSetResponderCapture={[Function onStartShouldSetResponderCapture]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0
          },
          null
        ]
      }>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flex": 1
          }
        }>
        <Text>
          TEST
        </Text>
      </View>
    </View>
  </View>
</View>
`;
